src/Raven.Server/ServerWide/OperationCancelToken.cs

```diff
@@ -15,19 +15,19 @@ public class OperationCancelToken : IDisposable
 
         public readonly CancellationToken Token;
 
-        public OperationCancelToken(TimeSpan cancelAfter, CancellationToken shutdown, CancellationToken requestAborted)
+        public OperationCancelToken(TimeSpan cancelAfter, params CancellationToken[] tokens)
```

Comment:
this is a bad idea, it creates a lot of allocations, it is better to have few ctors to handle 1,2,3 tokens